operator/v1/types_console.go

```diff
@@ -143,8 +143,133 @@ type Capability struct {
 	Visibility CapabilityVisibility `json:"visibility"`
 }
 
+// ThemeMode is the value of the logo theme mode that determines the theme mode in the console UI.
+// +kubebuilder:validation:Enum="Dark";"Light"
+// +enum
+type ThemeMode string
+
+// ThemeMode values
+const (
+	// ThemeModeDark represents the dark theme for the console UI.
+	ThemeModeDark ThemeMode = "Dark"
+
+	// ThemeModeLight represents the light theme for the console UI.
+	ThemeModeLight ThemeMode = "Light"
+)
+
+// LogoType is the value of the logo type that determines if the logo is for the masthead or the favicon in the console UI.
+// The masthead logo is displayed in the masthead and about modal of the console UI.
+// +kubebuilder:validation:Enum="Masthead";"Favicon"
+// +enum
+type LogoType string
+
+const (
+	// Masthead represents the logo in the masthead.
+	LogoTypeMasthead LogoType = "Masthead"
+
+	// Favicon represents the favicon logo.
+	LogoTypeFavicon LogoType = "Favicon"
+)
+
+// SourceType defines the source type of the file reference.
+// +kubebuilder:validation:Enum="ConfigMap"
+// +enum
+type SourceType string
+
+const (
+	// SourceTypeConfigMap represents a ConfigMap source.
+	SourceTypeConfigMap SourceType = "ConfigMap"
+)
+
+// ConfigMapFileReference references a specific file within a ConfigMap.
+type ConfigMapFileReference struct {
+	// name is the name of the ConfigMap.
+	// Must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character.
+	// Must be at most 253 characters in length.
+	// +kubebuilder:validation:MaxLength=253
+	// +kubebuilder:validation:XValidation:rule="!format.dns1123Subdomain().validate(self).hasValue()",message="a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character."
+	// +required
+	Name string `json:"name"`
+
+	// key is the logo key inside the referenced ConfigMap.
+	// Must consist only of alphanumeric characters, dashes (-), underscores (_), and periods (.).
+	// Must be at most 253 characters in length.
+	// +kubebuilder:validation:MaxLength=253
+	// +kubebuilder:validation:XValidation:rule="self.matches('.*\\\\.[a-zA-Z]{2,5}$')",message="The ConfigMap key must end with a valid file extension (2 to 5 letters)."
+	// +required
+	Key string `json:"key"`
+}
+
+type FileReferenceSource struct {
+	// from specifies the source type of the file reference.
+	// This field is an enum with a valid value "ConfigMap".
+	// +required
+	From SourceType `json:"from"`
+
+	// configMap specifies the ConfigMap source details.
+	// This field is required if From is "ConfigMap".
```

Comment:
Is there a CEL expression missing to enforce this?

Reply:
I think it should be something like:
`// +kubebuilder:validation:XValidation:rule="self.from == 'ConfigMap' || has(self.configMap)",message="configMap must be set when from is 'ConfigMap'."`

+ lets add an unit test for this case

Reply:
Probably looking for something like `+kubebuilder:validation:XValidation:rule="has(self.from) && self.from == 'ConfigMap' ? has(self.configMap) : !has(self.configMap)",message="configMap is required when from is ConfigMap, and forbidden otherwise"` which is the standard expression and message we use for discriminated unions

Reply:
What about this: `// +kubebuilder:validation:XValidation:rule="self.from == 'ConfigMap' ? self.configMap != null : self.configMap == null",message="ConfigMap must be set when From is 'ConfigMap'."
`

Reply:
I would recommend using the standard one I shared to stay consistent with the other discriminated unions throughout the openshift apis